File: scratch/inspect_db_real.py
import sqlite3
import os
import sys

sys.stdout.reconfigure(encoding='utf-8')

db_path = os.path.expandvars(r'%APPDATA%\البصمة الذكية\pos_data.db')
print("Connecting to database at:", db_path)

conn = sqlite3.connect(db_path)
cursor = conn.cursor()

try:
    cursor.execute("SELECT name FROM sqlite_master WHERE type='table'")
    tables = [t[0] for t in cursor.fetchall()]

    if 'business_settings' in tables:
        cursor.execute("SELECT key, value FROM business_settings")
        settings = cursor.fetchall()
        print("\n--- BUSINESS SETTINGS ---")
        for key, val in settings:
            print(f"{key}: {val}")
    
    if 'zatca_device' in tables:
        cursor.execute("SELECT id, device_id, csr_pem, compliance_csid, production_csid, private_key_pem FROM zatca_device")
        devices = cursor.fetchall()
        print("\n--- ZATCA DEVICE ---")
        for d in devices:
            print({
                'id': d[0],
                'device_id': d[1],
                'csr_pem': 'PRESENT' if d[2] else 'NULL',
                'compliance_csid': 'PRESENT' if d[3] else 'NULL',
                'production_csid': 'PRESENT' if d[4] else 'NULL',
                'private_key_pem': 'PRESENT' if d[5] else 'NULL',
            })
            if d[2]:
                print('CSR PEM:\n', d[2])

except Exception as e:
    print("Error:", e)
finally:
    conn.close()

File: scratch/inspect_db_real.js
const Database = require('better-sqlite3');
const path = require('path');
const os = require('os');

const userDataPath = path.join(os.homedir(), 'AppData', 'Roaming', 'البصمة الذكية');
const dbPath = path.join(userDataPath, 'pos_data.db');
console.log('Connecting to database at:', dbPath);

const db = new Database(dbPath);

try {
    const settings = db.prepare('SELECT * FROM settings').all();
    console.log('--- SETTINGS ---');
    settings.forEach(s => {
        console.log(`${s.key}: ${s.value}`);
    });

    const devices = db.prepare('SELECT * FROM zatca_devices').all();
    console.log('--- ZATCA DEVICES ---');
    devices.forEach(d => {
        console.log({
            id: d.id,
            device_id: d.device_id,
            csr_pem: d.csr_pem ? 'PRESENT' : 'NULL',
            compliance_csid: d.compliance_csid ? 'PRESENT' : 'NULL',
            production_csid: d.production_csid ? 'PRESENT' : 'NULL',
            private_key_pem: d.private_key_pem ? 'PRESENT' : 'NULL',
        });
        if (d.private_key_pem) {
            console.log('Private Key Start:', d.private_key_pem.substring(0, 50));
        }
    });
} catch (err) {
    console.error('Error reading DB:', err);
} finally {
    db.close();
}
